pkg/net/socketpair_unix.go

```diff
@@ -20,78 +20,27 @@ package net
 
 import (
 	"fmt"
-	"net"
 	"os"
+	"syscall"
 
-	syscall "golang.org/x/sys/unix"
+	"golang.org/x/sys/unix"
 )
 
-const (
-	local = 0
-	peer  = 1
-)
-
-// SocketPair contains the file descriptors of a connected pair of sockets.
-type SocketPair [2]int
-
 // NewSocketPair returns a connected pair of sockets.
 func NewSocketPair() (SocketPair, error) {
-	fds, err := syscall.Socketpair(syscall.AF_UNIX, syscall.SOCK_STREAM, 0)
-	if err != nil {
-		return [2]int{-1, -1}, fmt.Errorf("failed to create socketpair: %w", err)
-	}
-
-	return fds, nil
-}
-
-// LocalFile returns the socketpair fd for local usage as an *os.File.
-func (fds SocketPair) LocalFile() *os.File {
-	return os.NewFile(uintptr(fds[local]), fds.fileName()+"[0]")
-}
-
-// PeerFile returns the socketpair fd for peer usage as an *os.File.
-func (fds SocketPair) PeerFile() *os.File {
-	return os.NewFile(uintptr(fds[peer]), fds.fileName()+"[1]")
-}
-
-// LocalConn returns a net.Conn for the local end of the socketpair.
-func (fds SocketPair) LocalConn() (net.Conn, error) {
-	file := fds.LocalFile()
-	defer file.Close()
-	conn, err := net.FileConn(file)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create net.Conn for %s[0]: %w", fds.fileName(), err)
-	}
-	return conn, nil
-}
-
-// PeerConn returns a net.Conn for the peer end of the socketpair.
-func (fds SocketPair) PeerConn() (net.Conn, error) {
-	file := fds.PeerFile()
-	defer file.Close()
-	conn, err := net.FileConn(file)
+	syscall.ForkLock.RLock()
+	defer syscall.ForkLock.RUnlock()
+	fds, err := unix.Socketpair(unix.AF_UNIX, unix.SOCK_STREAM, 0)
```

Comment:
would you mind to use `unix.SOCK_CLOEXEC` here? so that we don't have to lock.

Reply:
unix.SOCK_CLOEXEC doesn't exists for darwin for example, this is the generic way

Reply:
I'm fine doing a linux only version, just tell me

Reply:
Let's focus on linux right now and add issue to track it for darwin. Thanks!

Reply:
I've added a linux and a unix version